docs/multiple-collectors.md

```diff
@@ -0,0 +1,166 @@
+| **test** | Identifies the act of benchmarking an *artifact* under a specific *test case*. Each test consists of several *test iterations*. |
+| **test iteration** | A single actual execution of a *test*. |
+| **collection** | A set of all *statistics* for a single *test iteration*. |
+| **test result** | The result of gathering all *statistics* from a single *test*. Aggregates results from all *test iterations* of that *test*, so a *test result* is essentially the union of *collections*. Usually we just take the minimum of each statistic out of all its *collections*. | 
+| **statistic** | A single measured value of a *metric* in a *test result*. |
+| **run** | A set of all *test results* for a set of *test cases* measured on a single *artifact*. |
+| **benchmark request** | A request for a benchmarking a *run* for a given *artifact*. Can be either created from a try build on a PR, or it is automatically determined from merged master/release *artifacts*. |
+| **benchmark set** | A selection of benchmarks that a collector will run. A collector is assigned a benchmark_set id |
+| **collector** | A physical runner for benchmarking the compiler. |
+| **cluster** | One or more collectors of the same target, for benchmarking the compiler. |
+| **collector_id** | A unique identifier of a *collector* (hard-coded at first for simplicity). |
+| **job** | High-level "work item" that defines a set of *test cases* that should be benchmarked on a specific collector. |
+| **job_queue** | Queue of *jobs*. |
+| **MAX_JOB_FAILS** | Maximum number of failures before a job is marked as a failed. |
+| **Assigning a job** | The act of allocating one or more *jobs* to a collector. |
+| **website** | A standalone server responsible for inserting work into the queue. |
+
+### Master artifacts
+
+The website will go through all recent master commits, and check if they are done by looking at the `sha` and `status` column in the `benchmark_request` table.
+
+- If the request is already marked as `completed`, nothing happens.
+- If the request is `in progress`, check [request completion](#Checking-request-completion).
+- If the request is `waiting for parent` commit benchmark to be completed, nothing happens.
+- If the request is missing, we will recursively find a set of parent master commits that are missing data (by looking at their status in `benchmark_request`).
+    - If the set is non-empty, these commits will be handled recursively with the same logic as this commit.
+    - If the set is empty, the request will be *enqueued*.
+
+### Try artifacts
+
+> The logic for try artifacts can either happen both in cron and in the GH webhook listener (that receives `@rust-timer queue/build` notifications), or only in cron.
+
+The website will go through all try artifacts in `benchmark_request` that are not yet marked as `completed`.
```

Comment:
In general, the text here is very pseudo-codey and doesn't correspond to how rustc-perf operates. It would be better to describe what the cron job does more literaly, I think.

Reply:
I feel I've done both, as attempt to explain the CRON more literally. I think this is quite useful, I've moved it to after the explanation of the CRON which feels like it flows a bit better now